pkg/controller/jobframework/reconciler.go

```diff
@@ -1116,12 +1116,8 @@ func (r *JobReconciler) ignoreUnretryableError(log logr.Logger, err error) error
 func generatePodsReadyCondition(job GenericJob, wl *kueue.Workload) metav1.Condition {
 	conditionStatus := metav1.ConditionFalse
 	message := "Not all pods are ready or succeeded"
-	// Once PodsReady=True it stays as long as the workload remains admitted to
-	// avoid unnecessary flickering the condition when the pods transition
-	// from Ready to Completed. As pods finish, they transition first into the
-	// uncountedTerminatedPods staging area, before passing to the
-	// succeeded/failed counters.
-	if workload.IsAdmitted(wl) && (job.PodsReady() || apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadPodsReady)) {
+
+	if workload.IsAdmitted(wl) && job.PodsReady() {
```

Comment:
Do I understand correctly the second check is currently equivalent to checking the condition? If so, then I would prefer to check the condition instead so that we have an additional buffer of logic. For example we may prefer to switch the condition to False only after a while when Pods ready responds false.

Also, is it true that even if PodsReady changes to False then the workload will not be automatically evicted?Keeping this by default is important because maybe there are some other Job integrations which could temporarily flip PodsReady to False. I think we want to be backwards compatible and keep the Jobs running. Unless the recovery timeout is specified

Reply:
> If so, then I would prefer to check the condition instead so that we have an additional buffer of logic

Ah I missed this is actually function responsible for the condition generation. LGTM.

Reply:
Oh, you're right, thanks for catching that. This PR was intended to be preparatory for the next I'm working on and I take into account this case, but for now it would change behavior. I've reverted the changes